File: MCTS_TicTacToe_python/TicTacToe.py
from Board import Board
from MonteCarloTreeSearch import MonteCarloTreeSearch

board = Board()

mcts1 = MonteCarloTreeSearch()
mcts3 = MonteCarloTreeSearch()

player = Board.P1

total_moves = Board.DEFAULT_BOARD_SIZE * Board.DEFAULT_BOARD_SIZE
for i in range(0, total_moves):
    if player == Board.P1:
        board = mcts3.find_next_move(board, player)
        print("Player 1")
    else:
        board = mcts1.find_next_move(board, player)
        print("Player 2")

    if board.check_status() != -1:
        break

    player = Board.P2 if player == Board.P1 else Board.P1

win_status = board.check_status()
print("Win status: " + str(win_status))
print(board)

File: MCTS_TicTacToe_java/src/com/baeldung/algorithms/mcts/tests/TicTacToe.java
package com.baeldung.algorithms.mcts.tests;

import com.baeldung.algorithms.mcts.montecarlo.MonteCarloTreeSearch;
import com.baeldung.algorithms.mcts.tictactoe.Board;

public class TicTacToe {
    public static void main(String[] args) {
        MonteCarloTreeSearch mcts = new MonteCarloTreeSearch();
        Board board = new Board();

        int player = Board.P1;
        int totalMoves = Board.DEFAULT_BOARD_SIZE * Board.DEFAULT_BOARD_SIZE;
        for (int i = 0; i < totalMoves; i++) {
            board = mcts.findNextMove(board, player);

            //System.out.println("Player: " + player);
            //board.printBoard();

            if (board.checkStatus() != -1) {
                break;
            }
            player = 3 - player;
        }
        int winStatus = board.checkStatus();

        System.out.println(winStatus);
    }
}
